pennylane/measure.py

```diff
@@ -152,35 +152,37 @@ def sample(op, n=None):
 
     Args:
         op (Observable): a quantum observable object
-        n (int): number of samples that should be obtained. Defaults to the
+        n (int): Number of samples that should be obtained. Defaults to the
             number of shots given as a parameter to the corresponding Device.
     """
     if not isinstance(op, Observable):
         raise QuantumFunctionError(
             "{} is not an observable: cannot be used with sample".format(op.name)
         )
 
-    if QNode._current_context is not None:
-        # delete operations from QNode queue
-        QNode._current_context.queue.remove(op)
-
-    # set return type to be a sample
-    op.return_type = "sample"
-
-    # decorate the operation with the number of samples specified
     if n is None:
         if QNode._current_context is not None:
-            op.num_samples = QNode._current_context.device.shots
-    else:
-        op.num_samples = n
+            n = QNode._current_context.device.shots
+        else:
+            raise QuantumFunctionError("Could not load number of samples from underlying device.")
```

Comment:
Yes. Exactly when `QNode._current_context` is `None`. I felt that I should raise an error if this was the case (because something is broken beyond repair when it is).

What change would you suggest here? Removing the error? Or also checking for the existence of the `.device`?

Reply:
I assumed another error would be raised before this one - an operation would fail at queuing (due to no device existing), so this error would never be accessible anyway.

Reply:
So should I remove it?

Reply:
My only suggestion would be to make the error message more clear. Something like "Could not find bound device to determine a default number of samples."